src/main/java/core/basesyntax/Car.java

```diff
@@ -1,13 +1,80 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 
-/**
- * Make this class immutable. See requirements in the README file
- */
-public class Car {
-    private int year;
-    private String color;
-    private List<Wheel> wheels;
-    private Engine engine;
+public final class Car {
+    private final int year;
+    private final String color;
+    private final List<Wheel> wheels;
+    private final Engine engine;
+
+    public Car(int year, String color, List<Wheel> wheels, Engine engine) {
+        this.year = year;
+        this.color = color;
+        List<Wheel> newWheels = new ArrayList<>();
+        for (Wheel wheel : wheels) {
+            newWheels.add(wheel.clone());
+        }
+        this.wheels = newWheels;
+        if (engine == null) {
+            this.engine = null;
+            return;
+        }
+        this.engine = engine.clone();
+    }
+
+    public int getYear() {
+        return year;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public List<Wheel> getWheels() {
+        return new ArrayList<>(wheels);
+    }
+
+    public Engine getEngine() {
+        if (engine == null) {
+            return null;
+        }
+        return engine.clone();
```

Comment:
```suggestion
        return engine == null ? null : engine.clone();
```
Let's make it easier.

Reply:
Changed